autoload/fzf/vim.vim

```diff
@@ -31,14 +31,18 @@ set cpo&vim
 let s:min_version = '0.23.0'
 let s:is_win = has('win32') || has('win64')
 let s:is_wsl_bash = s:is_win && (exepath('bash') =~? 'Windows[/\\]system32[/\\]bash.exe$')
+let s:is_defined_win_bash = get(g:, 'fzf_preview_win_bash', '') != ''
+if s:is_defined_win_bash
+  let s:defined_win_bash = split(system('for %A in ("'.g:fzf_preview_win_bash.'") do @echo %~sA'), "\n")[0]
+endif
```

Comment:
If I'm not mistaken `g:fzf_preview_win_bash` is only for Windows, so should we move this code block to line 45?

Reply:
I think we should use something like `s:bash` or `s:bash()` that returns the path to bash to use regardless of the platform and refer to it throughout the code instead of putting conditionals here and there.

Reply:
In fact, you can see in `line 44` is to determine whether it is running under windows (although only to determine the path of bash, but this path in other environments will not happen, so it should be safe), I changed `line 45` to this because in the case of the windows environment has been determined, if you call `c:\windows\ system32\bash.exe` to use, in the original `if` only `has ('nvim')` will lead to path errors (because `c:\windows\system32\bash.exe` actually requires the path parameters need to be wsl in the format, that is, else contained in the windows format will be amended to `/mnt/ blabla...` This format, so I made this `if`, if `s:is_defined_win_bash` judgment holds and is not custom `c:\windows\system32\bash.exe` case, we will continue to use the path under the windows rule. (This is also a defensive `if` on my part, as I ask in the do that if you want to customize bash, make sure the custom bash supports the path to the windows rule as an argument)
It's worth mentioning again that the operations here are based on the windows environment `if` on `line 44`, so there should be no conflict (at least in my tests so far)

and for the `s:bash`:
You're right, and I know I should, but as I mentioned above, I made a lot of poor `if else` because I wanted to make sure everywhere I could that it wouldn't interfere with running on other platforms besides windows.
Of course, if make a method is necessary, then I cound find some time to make it right  ~  : )
And also, obviously my English is very poor, I hope you can understand what I mean